mev_inspect/aave_liquidations.py

```diff
@@ -0,0 +1,128 @@
+            ):
+
+                liquidations.append(trace)
+                contract_addresses.append(trace.from_address)
+                tx.append(trace.transaction_hash)
+
+                # Found liquidation, now parse inputs for data
+                for input in trace.inputs:
+
+                    if input == "_purchaseAmount":
+                        liquidation_amount = trace.inputs[input]
+                    elif input == "_collateral":
+                        collateral_address = trace.inputs[input]
+                        # This will be the address of the sent token
+                    elif input == "_reserve":
+                        reserve = trace.inputs[input]
+                        # This will be the address of the received token
+                    elif input == "_user":
+                        liquidated_user = trace.inputs[input]
+                # Register liquidation
+                result.append(
+                    Liquidation(
+                        liquidated_user=liquidated_user,
+                        collateral_address=collateral_address,
+                        collateral_amount=liquidation_amount,
+                        collateral_source="",
+                        reserve=reserve,
+                    )
+                )
+
+            elif is_transfer_from_liquidator(trace, unique_transfer_hashes):
+
+                # Add the transfer
+                liquidator = next(
+                    (
+                        contract_addresses[i]
+                        for i in range(len(contract_addresses))
+                        if trace.inputs["sender"] == contract_addresses[i]
+                    ),
+                    "",
+                )
+
+                transfers_from.append(
+                    ["from", liquidator, trace.transaction_hash, trace.inputs["amount"]]
+                )
+                unique_transfer_hashes.append(trace.transaction_hash)
+
+            elif is_transfer_to_liquidator(trace):
+
+                # Add the transfer
+                liquidator = next(
+                    (
+                        contract_addresses[i]
+                        for i in range(len(contract_addresses))
+                        if trace.inputs["recipient"] == contract_addresses[i]
+                    ),
+                    "",
+                )
+                transfers_to.append(
+                    ["to", liquidator, trace.transaction_hash, trace.inputs["amount"]]
+                )
+
+    return [result, transfers_to, transfers_from]
+
+
+def is_transfer_from_liquidator(
+    trace: ClassifiedTrace,
+    unique_transfer_hashes: List[str],
+) -> bool:
+    """Check if transfer is from liquidator"""
+    transfer = ERC20Transfer.from_trace(trace)
+    if (
+        trace.classification == Classification.transfer
+        and transfer.from_address in contract_addresses
+        and trace.transaction_hash not in unique_transfer_hashes
+    ):
+        return True
```

Comment:
For both of these, any time you have
```
if x:
    return True
else:
    return False
```

it's the same as
```
return x
```

(same thing in reverse with a not)

Reply:
true, but the if is quite lengthy, this seems cleaner imo

Reply:
i think it’s actually less readable because it implies the conditional is more than a boolean somehow

wrapping a bool in an if return True is just reinforcing the existing bool’s value

it’s like me doing absolute_value(1) to make clear 1 is positive

Reply:
I think i's more readable:
is_c(c):
If (c): return True
else: return false

is a very formal way to check if condition c holds, the function follows this. Doesn't seem misleading to me